projects/storefrontlib/shared/components/captcha/captcha.component.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Output,
+  Renderer2,
+  ViewChild,
+} from '@angular/core';
+import { CaptchaService } from 'projects/storefrontlib/shared/components/captcha/captcha.service';
+import { CaptchaApiConfig } from './config/captcha-api-config';
+
+@Component({
+  selector: 'cx-captcha',
+  templateUrl: './captcha.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CaptchaComponent implements AfterViewInit {
+  @Output() confirmed = new EventEmitter<boolean>();
+  @Output() enabled = new EventEmitter<boolean>();
+
+  @ViewChild('captcha', { static: false }) captchaRef: ElementRef;
+
+  constructor(
+    protected config: CaptchaApiConfig,
+    protected captchaService: CaptchaService,
+    private renderer: Renderer2
+  ) {}
+
+  ngAfterViewInit(): void {
+    if (this.config?.fields) {
+      Object.keys(this.config.fields).forEach((key) => {
+        if (this.config.fields) {
```

Comment:
I think this is what was intended?
```suggestion
        if (this.config.fields[key]) {
```